assets/javascripts/discourse/initializers/apply-tag-autocomplete.js.es6

```diff
@@ -0,0 +1,55 @@
+import DEditor from 'discourse/components/d-editor';
+import ComposerEditor from 'discourse/components/composer-editor';
+import { categoryHashtagTriggerRule, SEPARATOR } from 'discourse/lib/category-hashtags';
+import { search as searchCategoryTag  } from 'discourse/plugins/discourse-tagging/lib/category-tag-search';
+import Category from 'discourse/models/category';
+import { fetchUnseenTagHashtags, linkSeenTagHashtags } from 'discourse/plugins/discourse-tagging/lib/link-tag-hashtag';
+import { TAG_HASHTAG_POSTFIX } from 'discourse/plugins/discourse-tagging/lib/tag-hashtags';
+import { linkSeenCategoryHashtags, fetchUnseenCategoryHashtags } from 'discourse/lib/link-category-hashtags';
+import { on } from 'ember-addons/ember-computed-decorators';
+
+export default {
+  name: 'apply-tag-autocomplete',
+
+  initialize(container, app) {
+    DEditor.reopen({
+      _applyCategoryHashtagAutocomplete() {
+        const template = container.lookup('template:category-tag-autocomplete.raw');
+
+        $('.d-editor-input').autocomplete({
+          template: template,
+          key: '#',
+          transformComplete(obj) {
+            if (obj.model) {
+              return Category.slugFor(obj.model, SEPARATOR);
+            } else {
+              return `${obj.text}${TAG_HASHTAG_POSTFIX}`;
+            }
+          },
+          dataSource(term) {
+            return searchCategoryTag(term);
+          },
+          triggerRule(textarea, opts) {
+            return categoryHashtagTriggerRule(textarea, opts);
+          }
+        });
+      }
+    });
+
+    ComposerEditor.reopen({
+      _renderUnseenTagHashtags($preview, unseen) {
+        fetchUnseenTagHashtags(unseen).then(() => {
```

Comment:
Small tip - if your funciton is one line you don't need curly braces and can just leave it on one line:

`fetchUnseenTagHashtags(unseen).then(() => linkSeenTagHashtags($preview));`